iroha/src/bridge.rs

```diff
@@ -1 +1,187 @@
+    /// Bridge owner's account Identification. Only this account will be able to manipulate the bridge.
+    pub owner_account_id: <Account as Identifiable>::Id,
+}
+
+impl Identifiable for BridgeDefinition {
+    type Id = BridgeDefinitionId;
+}
+
+/// Identification of a Bridge. Consists of Bridge's definition Identification.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Encode, Decode)]
+pub struct BridgeId {
+    /// Entity Identification.
+    definition_id: <BridgeDefinition as Identifiable>::Id,
+}
+
+impl BridgeId {
+    /// Default Bridge identifier constructor.
+    pub fn new(name: &str) -> Self {
+        BridgeId {
+            definition_id: BridgeDefinitionId::new(name),
+        }
+    }
+}
+
+/// An entity used for performing operations between Iroha and third-party blockchain.
+#[derive(Debug, Clone)]
+pub struct Bridge {
+    /// Component Identification.
+    id: <Bridge as Identifiable>::Id,
+    /// Bridge's account ID.
+    account_id: <Account as Identifiable>::Id,
+}
+
+impl Bridge {
+    /// Default `Bridge` entity constructor.
+    pub fn new(
+        id: <Bridge as Identifiable>::Id,
+        account_id: <Account as Identifiable>::Id,
+    ) -> Self {
+        Bridge { id, account_id }
+    }
+
+    /// A helper function for returning Bridge's name.
+    pub fn name(&self) -> &str {
+        &self.id.definition_id.name
+    }
+}
+
+impl Identifiable for Bridge {
+    type Id = BridgeId;
+}
+
+/// Iroha Special Instructions module provides extensions for `Peer` structure and an
+/// implementation of the generic `Register` Iroha Special Instruction for `Bridge` registration.
+pub mod isi {
+    use super::*;
+    use crate::isi::prelude::*;
+    use std::collections::HashMap;
+
+    impl Peer {
+        /// Get `Bridge`s without an ability to modify it.
+        pub fn read_bridges(&self) -> &HashMap<String, Bridge> {
+            &self.bridges
+        }
+
+        /// Get `Bridge`s with an ability to modify it.
+        pub fn bridges(&mut self) -> &mut HashMap<String, Bridge> {
+            &mut self.bridges
+        }
+
+        /// Constructor of `Register<Peer, BridgeDefinition>` Iroha Special Instruction.
+        pub fn register_bridge(
+            &self,
+            object: BridgeDefinition,
+        ) -> Register<Peer, BridgeDefinition> {
+            Register {
+                object,
+                destination_id: self.id.clone(),
+            }
+        }
+    }
+
+    impl Register<Peer, BridgeDefinition> {
+        /// Registers the `Bridge` by its definition on the given `WorldStateView`.
+        ///
+        /// Adds a `Domain` with the same name as in the `BridgeDefinition`, then registers an
+        /// account with default name "bridge" and a public key generated from a seed, and finally
+        /// constructs a `Bridge` entity from that data.
+        pub(crate) fn execute(self, world_state_view: &mut WorldStateView) -> Result<(), String> {
+            let bridge_def = self.object;
+
```

Comment:
Please do not use blank lines inside methods

Reply:
@humb1t isn't it harder to read? Anyway, fixed.

Reply:
it's space consuming, we should stick to a standard in formatting